Correction for Brazilian address format

TagaSanPedroAko opened the pull request:
Solves #11561

TagaSanPedroAko committed:
Add missing state field for Brazilian address format

tordans commented:
@TagaSanPedroAko do you have a reference like the wiki, forum, … that we can use to verify this. Or maybe some other reference like links to existing data… – this would make reviewing your PR easier and faster.

TagaSanPedroAko commented:
Here's a handy reference from the wiki (text in Portuguese): https://wiki.openstreetmap.org/wiki/Brasil/Padroniza%C3%A7%C3%A3o/Endere%C3%A7os.

Standard Brazilian postal addresses contain a state or more specifically, federative unit (*unidade da federação*, which includes both states and the lone Federal District containing Brasilia), written as a two-letter abbreviation, similar as with the US. The state is separated from the city in the address with a hyphen/en-dash. Looks like the present supported address format ended up without a state field for conflating the "city name - state/federative unit abbreviation" combination (such as São Paulo - SP, Belo Horizonte - MG, Curitiba - PR, Brasília - DF) as the whole city field.

TagaSanPedroAko committed:
correct location of postal code field for Brazilian address format

Corrects location of postal code for Brazilian address format, following standard postal address format used in Brazil as explained in https://wiki.openstreetmap.org/wiki/Brasil/Padroniza%C3%A7%C3%A3o/Endere%C3%A7os

TagaSanPedroAko commented:
I also noticed the placement of the postal code field is also incorrect as compared to the standard format explained in the wiki. I have included another change that corrects its placement and updated the connected issue description.

matkoniecz commented:
> Here's a handy reference from the wiki (text in Portuguese): https://wiki.openstreetmap.org/wiki/Brasil/Padroniza%C3%A7%C3%A3o/Endere%C3%A7os.

was it discussed with community anywhere?

Sadly OSM Wiki sometimes contains complete nonsense

matkoniecz commented:
@TagaSanPedroAko was it discussed with community at some point? Can you link that discussion?

matkoniecz commented:
@TagaSanPedroAko was it discussed with community at some point? Can you link that discussion?

TagaSanPedroAko commented:
@matkoniecz It has been discussed in the community forum: https://community.openstreetmap.org/t/padronizacao-de-enderecos-brasileiros
